Geometry/HcalTowerAlgo/src/HcalGeometry.cc

```diff
@@ -163,6 +163,23 @@ GlobalPoint HcalGeometry::getBackPosition(const DetId& id) const {
   }
 }
 
+CaloCellGeometry::CornersVec HcalGeometry::getCorners(const DetId& id) const {
+  if (!m_mergePosition) {
+    return (getGeometry(id)->getCorners());
+  } else {
+    std::vector<HcalDetId> ids;
+    m_topology.unmergeDepthDetId(HcalDetId(id),ids);
+    CaloCellGeometry::CornersVec mcorners;
+    CaloCellGeometry::CornersVec mcf = getGeometry(ids.front())->getCorners();
+    CaloCellGeometry::CornersVec mcb = getGeometry(ids.back())->getCorners();
+    for (unsigned int k=0; k<4; ++k) {
```

Comment:
All CaloCellGeometry defines a cornersvec with 8 elements. This is the definition of CaloCellGeometry::CornersVec. It does not need a further check to the best of my understanding